presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -1750,9 +1750,22 @@ private static Number toNumber(Object value)
         if (value instanceof Boolean) {
             return ((Boolean) value) ? 1 : 0;
         }
+        if (value instanceof String) {
+            return toBigDecimal((String) value);
```

Comment:
Here we have 2 exceptions paths: one here and one below.
This can be surprising in the future.

Let's change `toBigDecimal` to return Optional:

```
private static Optional<BigDecimal> toBigDecimal(String value)
{
    try {
        return Optional.of(new BigDecimal(value));
    }
    catch (NumberFormatException ne) {
        return Optional.empty();
    }
}
```

then the code here needs an `if`:
```
Optional<BigDecimal> bigDecimal = toBigDecimal((String) value);
if (bigDecimal.isPresent()) {
    return bigDecimal.get();
}
```

Reply:
fixed